### PR TITLE
Adding a tool to gather stats from a refflat file

### PR DESCRIPTION
**Pull request by ffinfo**

### CODE REVIEW

File: biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/refflatstats/RefflatStats.scala [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    htsjdk.samtools.util.Log
+      .setGlobalLogLevel(htsjdk.samtools.util.Log.LogLevel.valueOf(logger.getLevel.toString))
+
+    logger.info("Reading refflat file")
+
+    val geneReader = GeneAnnotationReader.loadRefFlat(
+      cmdArgs.refflatFile,
+      FastaUtils.getCachedDict(cmdArgs.referenceFasta))
+
+    val futures = geneReader.getAll.map(generateGeneStats(_, cmdArgs.referenceFasta)).toList
```
Comment (johnmous) [inline biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/refflatstats/RefflatStats.scala:73]: Spelling error? (features instead of futures)

Comment (ffinfo) [reply biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/refflatstats/RefflatStats.scala:73]: Referring to the object type, see also http://www.scala-lang.org/api/2.11.4/index.html#scala.concurrent.Future

File: biopet-utils/src/main/scala/nl/lumc/sasc/biopet/utils/FastaUtils.scala [hunk 2]
```diff
@@ -55,4 +55,22 @@ object FastaUtils {
...
+  }
+
+  /** This method returns the fraction of GC for a given region */
+  def getSequenceGc(referenceFile: IndexedFastaSequenceFile,
+                    contig: String,
+                    start: Long,
+                    end: Long): Double = {
+    require(referenceFile.isIndexed)
+    val sequence = referenceFile.getSubsequenceAt(contig, start, end)
+    val total = sequence.length()
```
Comment (johnmous) [inline biopet-utils/src/main/scala/nl/lumc/sasc/biopet/utils/FastaUtils.scala]: We calculated total bases based on the sum of g, c, a, t. So we excluded possible Ns from the total base counting.

**Review by johnmous — APPROVED** [review]

File: biopet-utils/src/main/scala/nl/lumc/sasc/biopet/utils/FastaUtils.scala (hunk 2 above)
Comment (ffinfo) [reply biopet-utils/src/main/scala/nl/lumc/sasc/biopet/utils/FastaUtils.scala]: ye your right, changed this

**Pull request merged**